contracts/PricerRule.sol

```diff
@@ -0,0 +1,408 @@
+     *          - The economy token address is not null.
+     *          - The base currency code is not empty.
+     *          - Conversion rate from the base currency to the token is not 0.
+     *          - Conversion rate's decimals from the base currency to the
+     *            token is not 0.
+     *          - The economy token rules address is not null.
+     *
+     * @param _organization Organization address.
+     * @param _eip20Token The economy token address.
+     * @param _baseCurrencyCode The economy base currency code.
+     * @param _conversionRate The conversion rate from the economy base currency
+     *                        to the token.
+     * @param _conversionRateDecimals The conversion rate's decimals from the
+     *                                economy base currency to the token.
+     * @param _tokenRules The economy token rules address.
+     */
+    constructor(
+        OrganizationInterface _organization,
+        address _eip20Token,
+        bytes3 _baseCurrencyCode,
+        uint256 _conversionRate,
+        uint8 _conversionRateDecimals,
+        address _tokenRules
+    )
+        Organized(_organization)
+        public
+    {
+        require(_eip20Token != address(0), "EIP20 Token address is null.");
+
+        require(
+            _baseCurrencyCode != bytes3(0),
+            "Base currency code is null."
+        );
+
+        require(
+            _conversionRate != 0,
+            "Conversion rate from the base currency to the token is 0."
+        );
+
+        require(
+            _conversionRateDecimals != 0,
+            "Conversion rate's decimals from the base currency to the token is 0."
+        );
+
+        require(
+            _tokenRules != address(0),
+            "Token rules address is null."
+        );
+
+        eip20Token = EIP20TokenInterface(_eip20Token);
+
+        conversionRateFromBaseCurrencyToToken = _conversionRate;
+
+        conversionRateDecimalsFromBaseCurrencyToToken = _conversionRateDecimals;
+
+        tokenRules = TokenRules(_tokenRules);
+    }
+
+
+    /* External Functions */
+
+    /**
+     * @notice Transfers from the msg.sender account to the specified addresses
+     *         an amount of the economy tokens equivalent (after conversion) to
+     *         the specified amounts in pay currency.
+     *
+     * @dev Function requires:
+     *          - The lengths of arrays of beneficiaries and amounts are equal.
+     *          - The intended price of the base currency in the pay currency
+     *            wrt the current price is in the registered acceptance margin.
+     *
+     * @param _payCurrencyCode Currency code of the specified amounts.
+     * @param _baseCurrencyIntendedPrice The intended price of the base currency
+     *                                   used during conversion within function.
+     */
+    function pay(
+        bytes3 _payCurrencyCode,
+        uint256 _baseCurrencyIntendedPrice,
+        address[] _toList,
```

Comment:
Is commission gone or is it that it should be represented, when present, by another to/amount pair in the arrays (i.e., just another transfer)?

Reply:
Second option. We've agreed that commission calculation (if it's percentage for example) is done off-chain and on-chain we just have another to/amount pair if there is a beneficiary and beneficiary amount.

Reply:
Super 🎉